pkg/cmd/server/bootstrappolicy/controller_policy.go

```diff
@@ -93,6 +97,36 @@ func init() {
 			eventsRule(),
 		},
 	})
+
+	// origin-namespace-controller
+	addControllerRole(rbac.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{Name: saRolePrefix + InfraOriginNamespaceServiceAccountName},
+		Rules: []rbac.PolicyRule{
+			rbac.NewRule("get", "list", "watch").Groups(kapiGroup).Resources("namespaces").RuleOrDie(),
+			rbac.NewRule("update").Groups(kapiGroup).Resources("namespaces/finalize", "namespaces/status").RuleOrDie(),
+			eventsRule(),
+		},
+	})
+
+	// serviceaccount-controller
+	addControllerRole(rbac.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{Name: saRolePrefix + InfraServiceAccountControllerServiceAccountName},
+		Rules: []rbac.PolicyRule{
+			rbac.NewRule(ReadWrite...).Groups(kapiGroup).Resources("serviceaccounts").RuleOrDie(),
```

Comment:
We don't use these in the controller_policy upstream.  Mutation is rarely uniform in the controllers and the powers are subtly different.  Update can require knowledge of resourceversion, patch doesn't.  Delete requires knowledge of names, deletecollection does not.